test/schema.validation.test.js

```diff
@@ -564,6 +564,44 @@ describe('schema', function() {
           done();
         }, { a: 'b' });
       });
+
+      it('doValidateSync should ignore async function and script waiting for promises (gh-4885)', function(done) {
+        let asyncCalled = false;
+        let normalCalled = false;
+        let promiseCalled = false;
+        let promiseCompleted = false;
+
+        const schema = new Schema({
+          prop: {
+            type: Boolean,
+            validate: [
+              {
+                validator: async() => { asyncCalled = true; }
+              },
+              {
+                validator: () => { normalCalled = true; }
+              },
+              {
+                validator: () => {
+                  promiseCalled = true;
+                  return new Promise((res) => setTimeout(() => {
```

Comment:
I think we can convert this whole promise to `Promise.resolve();`, no need for a setTimeout.
Also, if async validators don't work, we should throw an error if we receive a validator that returns a promise, similar to https://github.com/Automattic/mongoose/pull/9176.
What do you think? @hasezoey

Reply:
> Also, if async validators don't work, we should throw an error if we receive a validator that returns a promise, similar to

from what i know async validators are allowed and in the documentation and also executed in `validate`, but just not in `validateSync` (stated in the documentation)

Reply:
Interesting, I didn't know that.
So they would work with `await Document#validate()`, but not with `Document#validateSync()` nor implicitly when running `await user.save();`, is that about right? @hasezoey

Reply:
> I think we can convert this whole promise to Promise.resolve();, no need for a setTimeout.

i made it this way because promises may be executed earlier when no "blocking" is found, and i wanted to check that promise function get called but "not finish executing" and `Promise.resolve` would "maybe immediately" set this value

Reply:
> Interesting, I didn't know that. So they would work with `await Document#validate()`, but not with `Document#validateSync()` nor implicitly when running `await user.save();`, is that about right? @hasezoey

from what i know, in both `validate` and `save` async validators are run (and promises "awaited") (from what i can tell, `save` calls `validate`)

Reply:
Makes sense. Thanks @hasezoey